src/main/java/ch/njol/skript/effects/EffLog.java

```diff
@@ -77,42 +78,48 @@ public void close() {
 	private Expression<String> messages;
 	@Nullable
 	private Expression<String> files;
-	
+
+	private Level logLevel = Level.INFO;
+
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
 		messages = (Expression<String>) exprs[0];
 		files = (Expression<String>) exprs[1];
+		if (parser.mark == 1) {
+			logLevel = Level.WARNING;
+		} else if (parser.mark == 2) {
+			logLevel = Level.SEVERE;
+		}
 		return true;
 	}
 	
 	@SuppressWarnings("resource")
 	@Override
-	protected void execute(final Event e) {
-		for (final String message : messages.getArray(e)) {
+	protected void execute(Event event) {
+		for (String message : messages.getArray(event)) {
 			if (files != null) {
-				for (String s : files.getArray(e)) {
-					s = s.toLowerCase(Locale.ENGLISH);
-					if (!s.endsWith(".log"))
-						s += ".log";
-					if (s.equals("server.log")) {
-						SkriptLogger.LOGGER.log(Level.INFO, message);
-						continue;
+				for (String logFile : files.getArray(event)) {
+					logFile = logFile.toLowerCase(Locale.ENGLISH);
+					if (!logFile.endsWith(".log"))
+						logFile += ".log";
+					if (logFile.equals("server.log")) {
+						SkriptLogger.LOGGER.log(logLevel, message);
 					}
-					PrintWriter w = writers.get(s);
-					if (w == null) {
-						final File f = new File(logsFolder, s); // REMIND what if s contains '..'?
+					PrintWriter logWriter = writers.get(logFile);
+					if (logWriter == null) {
+						File logFolder = new File(logsFolder, logFile); // REMIND what if logFile contains '..'?
 						try {
-							f.getParentFile().mkdirs();
-							w = new PrintWriter(new BufferedWriter(new FileWriter(f, true)));
-							writers.put(s, w);
-						} catch (final IOException ex) {
-							Skript.error("Cannot write to log file '" + s + "' (" + f.getPath() + "): " + ExceptionUtils.toString(ex));
+							logFolder.getParentFile().mkdirs();
+							logWriter = new PrintWriter(new BufferedWriter(new FileWriter(logFolder, true)));
+							writers.put(logFile, logWriter);
+						} catch (IOException ex) {
```

Comment:
From what I can see, if an IOException occurs, the method immediately returns and won't write to any remaining logs that might still be in queue. This may be intentional though 🤷‍♂️

Reply:
Out of the scope of the PR. I'm just simply adding severities. There was also talk of this Effect getting deleted